### ISSUE TITLE
--version flag

### ISSUE THREAD

**Issue by montyz** [kubernetes-sigs/aws-iam-authenticator#125]
On a support chat with AWS they asked for the version of `heptio-authenticator-aws` and it doesn't seem to have a flag to report the version of itself. It would be helpful to have that.

**Comment by christopherhein** [comment]
I like this idea and I might recommend we do this as a subcommand instead. I have a very simple to use package I use for other projects for returning the version of a `goreleaser` and `cobra` based project (not that this is difficult but it just adds some additional mutations around versioning for CI processes)

https://github.com/christopherhein/go-version

**Comment by nckturner** [comment]
I like the subcommand idea too.  Glanced at your package @christopherhein and it looks pretty cool, the only thing I don't agree with is using dates in binaries.  This hampers reproducible builds.

**Comment by christopherhein** [comment]
Happy to take a stab at this, if you want. About the dates, I kind of understand what you mean. I built this modeling after the way that `kubectl` works as well as the default `ldflags` that `goreleaser` uses - https://goreleaser.com/#using-the-main-version

In terms of reproducibility wouldn't we want binaries to have different dates if they were changed? that way we can have even more fidelity when version checking capabilities? At least for our published binaries.

`kubectl version` output:

```bash
→ kubectl version --client
Client Version: version.Info{Major:"1", Minor:"11", GitVersion:"v1.11.1", GitCommit:"b1b29978270dc22fecc592ac55d903350454310a", GitTreeState:"clean", BuildDate:"2018-07-18T11:36:43Z", GoVersion:"go1.10.3", Compiler:"gc", Platform:"darwin/amd64"}
```

The difference with mine being JSON default instead of a golang struct and a shortened if you pass a flag.
